consult.el

```diff
@@ -1044,6 +1044,55 @@ The arguments and expected return value are as specified for
         (funcall exit completion exit-status))
       t)))
 
+(defun consult--extract-commands (mode)
+  "Extract commands from MODE."
+  (let ((library-path (symbol-file mode))
+        (mode-rx (regexp-quote
+                  (substring (if (eq major-mode 'c-mode)
+                                 "cc-mode"
+                               (symbol-name major-mode))
+                             0 -5))))
+    (mapcan
+     (lambda (feature)
+       (let ((path (car feature)))
+         (when (or (string= path library-path)
+                   (string-match-p mode-rx (file-name-nondirectory path)))
+           (mapcar #'cdr
+                   (seq-filter
+                    (lambda (item)
+                      (and (consp item)
+                           (eq (car item) 'defun)
+                           (commandp (cdr item))
+                           (not (string-match-p "--" (symbol-name cmd)))))
+                    (cdr feature))))))
+     load-history)))
+
+;;;###autoload
+(defun consult-major-command ()
+  "Run a command from the current major mode."
+  (interactive)
+  (call-interactively
+   (consult--read "M-X: "
+                  (consult--extract-commands major-mode)
+                  :require-match t
+                  :category 'command
+                  :lookup #'intern)))
+
+;;;###autoload
+(defun consult-minor-command ()
+  "Run a command from one of the current active minor modes."
+  (interactive)
+  (call-interactively
+   (consult--read "M-X: "
+                  (mapcan #'consult--extract-commands
+                          (seq-filter (lambda (m)
+                                        (and (boundp m) (symbol-value m)))
+                                      minor-mode-list))
```

Comment:
Here my question is if it makes sense to distinguish local and global modes. We could use narrowing! We could make it one command `consult-mode-command` and then narrow with some keys for minor local/minor global/major. I would probably prefer that over having distinct commands.

Reply:
A single command with narrowing does sound good.

Reply:
Ok, let's do this then. I think it is a good idea - mostly people want all commands I gues. But if you want to explore you can narrow.

Reply:
We could also think about adding another narrowing possibility to only show commands with bindings, this goes in the direction of consult-binding...

Reply:
I didn't add the binding stuff.